test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_arguments.py

```diff
@@ -75,3 +63,8 @@ def Q(X): return expgrad._pmf_predict(X)[:, 1]
         assert error == pytest.approx(0.25, abs=_PRECISION)
         assert expgrad._n_oracle_calls == 32
         assert n_predictors == 3
+
+        # ensure that the input data wasn't changed by our mitigator
+        assert isinstance(transformed_X, type(transformX(X)))
+        assert isinstance(transformed_y, type(transformY(y)))
+        assert isinstance(transformed_A, type(transformA(A)))
```

Comment:
I don't think that these three added asserts will ever throw an exception. Is the intention to ensure that we don't apply some validation routine that will turn a list into a Series or a DataFrame? I don't think we would detect that here. That's because python passes around either immutable types or pointers, but there's no calling by reference.

Reply:
Yes, not sure how I thought that would work. There's a different way to test that, though. Let me rewrite this.

Reply:
I changed this. Instead I'm now checking for the data type of the args to estimator.fit(...)
LMK what you think!